addons/website_forum/static/src/js/website_forum.js

```diff
@@ -246,11 +246,12 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
             msg = _t("Sorry you must be logged in to perform this action");
             title = _t("Access Denied");
         }
-        this.call('crash_manager', 'show_warning', {
+        this.displayNotification({
             message: msg,
             title: title,
-        }, {
             sticky: false,
+            type: "warning",
+            messageIsHtml: true,
```

Comment:
can't we use IsHtml only when we enter in L243 ? And that we are sure about the input ?

Reply:
Exactly this, the message is currently safe(ish) assuming `karma` and `forum_id` can't be reflected, but it'd be better to make the entire thing safe by construction:

* only key in `messageIsHtml` if we actually need HTML injection
* properly escape all the non-static data

So something along the lines of:
```js
        var attrs = {};
        if (forum_id) {
            msg = _.str.sprintf('%s<a class="alert-link" href="/forum/%d/faq">%s</a>',
                _.escape(msg),
                forum_id, // or `%s` and `_.escape` it if it's may be something more than just an id (e.g. a slug)
                _.escape(_t("Read the guidelines to know how to gain karma."))
             );
            attrs = { messageIsHtml: true };
        }
        if (session.is_website_user) {
            msg = _t("Sorry you must be logged in to perform this action");
            title = _t("Access Denied");
        }
        this.displayNotification({
            message: msg,
            title: title,
            sticky: false,
            type: "warning",
            ...attrs,
        });
```

Reply:
> Just curious: How are we more unsure of the input if we enter L243?  If anything, that's the opposite?

in case one day we don't strip data-* on forum e.g.
Anybody will be able to add a <a href='#' class='karma_required' data-karma='xss'>link</a>, and you will alert a 'xss' unexpected message instead of a integer. So because we know in this case that we don't need html, it is better imo to remove this risk for the future. 

As confirmed by @xmo-odoo, ty for your review ;)